src/west/app/project.py

```diff
@@ -714,22 +711,57 @@ def has_checked_out_branch(project):
                         self._has_nonempty_status(project)):
                     continue
 
-                self.banner(f'{project.name_and_path}:')
-                # `git status` shows `manifest-rev` "sometimes", see
-                # #643. If manifest-rev is missing, then we already
-                # failed earlier anyway.
-                self.small_banner('manifest-rev:')
-                project.git('log --oneline -n 1 manifest-rev')
-                self.small_banner('status:')
-                project.git('status')
-                printed_status = True
+                self.compare(project)
+                printed_output = True
             except subprocess.CalledProcessError:
                 failed.append(project)
         self._handle_failed(args, failed)
 
-        if args.exit_code and printed_status:
+        if args.exit_code and printed_output:
             raise CommandError(1)
 
+    def compare(self, project):
+        self.banner(f'{project.name_and_path}:')
+        self.print_rev_info(project)
+        self.print_status(project)
+
+    def print_rev_info(self, project):
+        # For the manifest repository, unconditionally print
+        # HEAD's SHA and commit title.
+        #
+        # For other repositories, print HEAD's and manifest-rev's
+        # SHAs and commit titles, but only if they are different.
+        #
+        # We force git not to print in color to reduce noise in the
```

Comment:
```suggestion
        # We force git log not to print in color so west colored separators stand out
        # more.
```
git status is still colored by default.